File: stringLengthDemo/stringLengthDemo/ViewController.m
//
//  ViewController.m
//  stringLengthDemo
//
//  Created by 张艺哲 on 2019/7/29.
//  Copyright © 2019 张艺哲. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [ViewController testWithText: @"1你"];
//    [ViewController testWithText: @"123456"];
//    [ViewController testWithText: @"这15762是一段asdfsafdaf 次😂ˉˉ"];
//    [ViewController testWithText: @"这1576💰🙅‍♂️😸♋️💰🎃🤖🈶️🙅‍♂️🖐🤦‍♂️"];
    
}

+ (NSInteger) testWithText : (NSString *) text
{
    
    NSLog(@"%zd", text.length);
    
     NSLog(@"%zd", [[text dataUsingEncoding:CFStringConvertEncodingToNSStringEncoding(kCFStringEncodingGB_18030_2000)] length]);

    return text.length;
}


@end

File: TimeConverter/TimeConverter/ViewController.swift
//
//  ViewController.swift
//  TimeConverter
//
//  Created by Yizhe.Zhang on 2019/3/19.
//  Copyright © 2019 com.yizheyun. All rights reserved.
//

import UIKit
import SwiftyTimer

class ViewController: UIViewController {
    
    @IBOutlet weak var timeDisplayLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        Timer.every(1) {
            //当前时间戳
            let timeInterval:TimeInterval = self.datePicker.date.timeIntervalSince1970
            let timeStamp = Int(timeInterval)
            
            self.timeDisplayLabel.text = TimeConverterModel().TimeConverter(InputTimestap: timeStamp)
        }

    }
    

    
    @IBOutlet weak var datePicker: UIDatePicker!
    @IBAction func datePickerChange(_ sender: UIDatePicker) {
        
        //当前时间戳
        let timeInterval:TimeInterval = sender.date.timeIntervalSince1970
        let timeStamp = Int(timeInterval)
        
        timeDisplayLabel.text = TimeConverterModel().TimeConverter(InputTimestap: timeStamp)
    }
    
}

